runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/endpoints/EndpointProvider.kt

```diff
@@ -5,19 +5,14 @@
 
 package aws.smithy.kotlin.runtime.http.endpoints
 
-import aws.smithy.kotlin.runtime.util.InternalApi
-
 /**
  * Resolves endpoints for a given service client.
- *
- * This is a static version of the otherwise-generated endpoint resolver interface and is intended for use by internal
- * clients within the runtime.
  */
-@InternalApi
-public fun interface EndpointResolver {
+public fun interface EndpointProvider<T> {
```

Comment:
**Question**: Is there a semantic difference between a "provider" and a "resolver"? This interface's name changed but its KDoc and single method name still refer to "resolution" rather than "providing".

Reply:
I don't think there is, I preferred resolver with a single `resolve` method but this matches the SEP/ref arch terminology I believe.